example-extension/.github/workflows/ci.yml

```diff
@@ -1,73 +1,78 @@
-        run: 'composer install --no-progress'
-      - name: 'Run command'
-        run: 'composer ci:${{ matrix.command }}'
-    strategy:
-      fail-fast: false
-      matrix:
-        command:
-          - 'composer:normalize'
-          - 'composer:psr-verify'
-          - 'json:lint'
-          - 'php:rector'
-          - 'php:cs-fixer'
-          - 'php:stan'
-        php-version:
-          - '8.1'
-          - '8.2'
+    php-lint:
+        name: 'PHP linter'
+        runs-on: ubuntu-22.04
+        steps:
+            -
+                name: Checkout
+                uses: actions/checkout@v4
+            -
+                name: 'Install PHP'
+                uses: shivammathur/setup-php@v2
+                with:
+                    php_version: '${{ matrix.php_version }}'
+                    coverage: none
+                    tools: 'composer:v2.4'
+            -
+                name: 'Show the Composer configuration'
+                run: 'composer config --global --list'
+            -
+                name: 'Run PHP lint'
+                run: 'composer ci:php:lint'
+        strategy:
+            fail-fast: false
+            matrix:
+                php_version:
+                    - 8.1
+                    - 8.2
+    code-quality:
+        name: 'Code quality checks'
+        runs-on: ubuntu-22.04
+        steps:
+            -
+                name: Checkout
+                uses: actions/checkout@v4
+            -
+                name: 'Install PHP'
+                uses: shivammathur/setup-php@v2
+                with:
+                    php_version: '${{ matrix.php_version }}'
+                    coverage: none
+                    tools: 'composer:v2.4'
+            -
+                name: 'Show Composer version'
+                run: 'composer --version'
+            -
+                name: 'Show the Composer configuration'
+                run: 'composer config --global --list'
+            -
+                name: 'Cache dependencies installed with composer'
+                uses: actions/cache@v3
+                with:
+                    key: 'php${{ matrix.php_version }}-typo3${{ matrix.php_version }}'
+                    path: ~/.cache/composer
+                    restore-keys: "php${{ matrix.php_version }}-composer-\n"
+            -
+                name: 'Install Composer dependencies'
+                run: "composer require typo3/cms-core ^${{ matrix.typo3_version }} --no-progress\ncomposer install --no-progress\n"
+            -
+                name: 'Run command'
+                run: "composer ci:composer:psr-verify\ncomposer ci:json:lint\ncomposer ci:php:rector\ncomposer ci:php:cs-fixer\ncomposer ci:php:stan\n"
+        strategy:
+            fail-fast: false
+            matrix:
+                php_version:
+                    - 8.1
+                    - 8.2
+                typo3_version:
+                    - 11.5
+                    - 12.4
```

Comment:
@mkarulin Currently we do not run any tests in the pipelines, right? Can change that? I think it is possible to setup the pipeline so that tests are run if they exist but it does not fail if no test files are present.

In addition we would probably have to make sure that the `.gitkeep` file is created for both folders `Tests/Functional` and `Tests/Unit` if the folders are empty and if the file does not exist yet.

I think we did this already for the plain PHP code quality package, though the tests there are different from our TYPO3 tests here.